src/audio/dai.c

```diff
@@ -139,6 +139,8 @@ static struct comp_dev *dai_new(struct sof_ipc_comp *comp)
 	if (!dev)
 		return NULL;
 
+	dev->size = COMP_SIZE(struct sof_ipc_comp_dai);
```

Comment:
What about variable arrays in some of these IPCs ?

Reply:
`comp_dev` structures are allocated using size calculated from this macro, so I assume it's working OK.